### PR TITLE
docs: add documentation on `getCachedData` option

### LINKED ISSUE #22622: Clarify that `useAsyncData` results are not cached
### Environment

------------------------------
- Operating System: Linux
- Node Version:     v18.16.1
- Nuxt Version:     3.6.1
- Nitro Version:    2.5.2
- Package Manager:  yarn@1.22.19
- Builder:          vite
- User Config:      runtimeConfig, alias, css, components, vite, dir, plugins, modules, pwa
- Runtime Modules:  nuxt-jsonld@2.0.8, @vite-pwa/nuxt@0.1.0, @pinia/nuxt@0.4.11, @nuxtjs/eslint-module@4.1.0
- Build Modules:    -
------------------------------

### Reproduction

https://stackblitz.com/edit/github-tex5o8?file=pages%2Findex.vue

### Describe the bug

I've checked docs and as I see `useAsyncData` should cache data for response with same key.

So for example:
```
const KEY = 'test';
const { data: data } = await useAsyncData(KEY, async () => {
  console.log('test');
  return {};
});
const { data: data2 } = await useAsyncData(KEY, async () => {
  console.log('test');
  return {};
});
const { data: data3 } = await useAsyncData(KEY, async () => {
  console.log('test');
  return {};
});
```

This should show only once `'test'` in console, but it shows it three times.

In real world app I need to fetch data in component A and then use same data in page B.
Right now the only option that I see is to make two api calls.

In nuxt 2, I've used `fetch` component option method to make call in component A and I didn't even need to make call in page B, because I've already had data in vuex (from component A). How to handle that (without duplicated requests) if `useAsyncData` doesn't provide cache?

### Additional context

_No response_

### Logs

_No response_

### LINKED ISSUE #24694: Add `getCachedData` to `useFetch` and `useAsyncData` docs
### Describe the feature

The docs pages for [useFetch](https://nuxt.com/docs/api/composables/use-fetch#params) and [useAsyncData](https://nuxt.com/docs/api/composables/use-async-data#params) are missing the `getCachedData` parameter. 

I'll open a PR to add it to the docs. I would also like to add a link to the [Client-side Caching with getCachedData](https://youtu.be/aQPR0xn-MMk) example by @manniL if that's OK?

Related: https://github.com/nuxt/nuxt/issues/24271

### Additional information

- [X] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

### PR DESCRIPTION
**Pull request by luc122c**
### 🔗 Linked issue

Resolves #24694, resolves https://github.com/nuxt/nuxt/issues/22622

### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Adds missing `useCachedData` param to `useFetch` and `useAsyncData` composable docs. Also adds a callout linking to an example video by @manniL 

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by luc122c** [comment]
Related: https://github.com/nuxt/nuxt/issues/22622

Do you think this change satisfies this issue, or is it worth adding some additional notes?

File: docs/3.api/2.composables/use-async-data.md [hunk 1]
```diff
@@ -61,6 +61,7 @@ const { data: posts } = await useAsyncData(
   - `immediate`: when set to `false`, will prevent the request from firing immediately. (defaults to `true`)
   - `default`: a factory function to set the default value of the `data`, before the async function resolves - useful with the `lazy: true` or `immediate: false` option
   - `transform`: a function that can be used to alter `handler` function result after resolving
+  - `useCachedData`: a function which returns cached data. A *nullish* return will trigger a fetch.
```
Comment (manniL) [inline docs/3.api/2.composables/use-async-data.md:64]: Might be good to mention the default somewhere ☺️

Comment (luc122c) [reply docs/3.api/2.composables/use-async-data.md:64]: The default? As in, returning data in this function returns that as the data?

Comment (luc122c) [reply docs/3.api/2.composables/use-async-data.md]: Please see what you think of the updated wording 👍

File: docs/3.api/2.composables/use-async-data.md [hunk 2]
```diff
@@ -61,6 +61,7 @@ const { data: posts } = await useAsyncData(
   - `immediate`: when set to `false`, will prevent the request from firing immediately. (defaults to `true`)
   - `default`: a factory function to set the default value of the `data`, before the async function resolves - useful with the `lazy: true` or `immediate: false` option
   - `transform`: a function that can be used to alter `handler` function result after resolving
+  - `useCachedData`: By default, no caching happens. Provide a function which returns cached data. A *nullish* return will trigger a fetch.
```
Comment (danielroe) [inline docs/3.api/2.composables/use-async-data.md:64]:
```suggestion
  - `getCachedData`: By default, no caching happens. Provide a function which returns cached data. A *nullish* return will trigger a fetch.
```

File: docs/3.api/2.composables/use-fetch.md [hunk 3]
```diff
@@ -112,7 +117,7 @@ If you provide a function or ref as the `url` parameter, or if you provide funct
 By default, Nuxt waits until a `refresh` is finished before it can be executed again.
 
 ::callout
-If you have not fetched data on the server (for example, with `server: false`), then the data _will not_ be fetched until hydration completes. This means even if you await `useFetch` on client-side, `data` will remain null within `<script setup>`.
+If you have not fetched data on the server (for example, with `server: false`), then the data *will not* be fetched until hydration completes. This means even if you await `useFetch` on client-side, `data` will remain null within `<script setup>`.
```
Comment (danielroe) [inline docs/3.api/2.composables/use-fetch.md:120]:
```suggestion
If you have not fetched data on the server (for example, with `server: false`), then the data _will not_ be fetched until hydration completes. This means even if you await `useFetch` on client-side, `data` will remain null within `<script setup>`.
```

File: docs/3.api/2.composables/use-async-data.md [hunk 4]
```diff
@@ -84,7 +89,7 @@ You can use `useLazyAsyncData` to have the same behavior as `lazy: true` with `u
 By default, Nuxt waits until a `refresh` is finished before it can be executed again.
 
 ::callout
-If you have not fetched data on the server (for example, with `server: false`), then the data _will not_ be fetched until hydration completes. This means even if you await [`useAsyncData`](/docs/api/composables/use-async-data) on the client side, `data` will remain `null` within `<script setup>`.
+If you have not fetched data on the server (for example, with `server: false`), then the data *will not* be fetched until hydration completes. This means even if you await [`useAsyncData`](/docs/api/composables/use-async-data) on the client side, `data` will remain `null` within `<script setup>`.
```
Comment (danielroe) [inline docs/3.api/2.composables/use-async-data.md:92]:
```suggestion
If you have not fetched data on the server (for example, with `server: false`), then the data _will not_ be fetched until hydration completes. This means even if you await [`useAsyncData`](/docs/api/composables/use-async-data) on the client side, `data` will remain `null` within `<script setup>`.
```

File: docs/3.api/2.composables/use-async-data.md [hunk 5]
```diff
@@ -61,6 +61,7 @@ const { data: posts } = await useAsyncData(
   - `immediate`: when set to `false`, will prevent the request from firing immediately. (defaults to `true`)
   - `default`: a factory function to set the default value of the `data`, before the async function resolves - useful with the `lazy: true` or `immediate: false` option
   - `transform`: a function that can be used to alter `handler` function result after resolving
+  - `getCachedData`: By default, no caching happens. Provide a function which returns cached data. A *nullish* return will trigger a fetch.
```
Comment (danielroe) [inline docs/3.api/2.composables/use-async-data.md:64]:
```suggestion
  - `getCachedData`: Provide a function which returns cached data. A *nullish* return will trigger a fetch. By default, this is: `key => nuxt.isHydrating ? nuxt.payload.data[key] : nuxt.static.data[key]`.
```

File: docs/3.api/2.composables/use-fetch.md [hunk 6]
```diff
@@ -93,6 +93,7 @@ All fetch options can be given a `computed` or `ref` value. These will be watche
   - `immediate`: when set to `false`, will prevent the request from firing immediately. (defaults to `true`)
   - `default`: a factory function to set the default value of the `data`, before the async function resolves - useful with the `lazy: true` or `immediate: false` option
   - `transform`: a function that can be used to alter `handler` function result after resolving
+  - `getCachedData`: By default, no caching happens. Provide a function which returns cached data. A *nullish* return will trigger a fetch.
```
Comment (danielroe) [inline docs/3.api/2.composables/use-fetch.md:96]:
```suggestion
  - `getCachedData`: Provide a function which returns cached data. A _nullish_ return will trigger a fetch. By default, this is: `key => nuxt.isHydrating ? nuxt.payload.data[key] : nuxt.static.data[key]`, which only caches data when `payloadExtraction` is enabled.
```

**Review by danielroe — APPROVED** [review]

**Comment by luc122c** [comment]
Oops! Didn't spot the typo! Thanks 👍

The auto fix CI keeps changing the italics markdown.

**Pull request merged**